arangod/Network/Methods.cpp

```diff
@@ -178,148 +176,149 @@ FutureRes sendRequest(ConnectionPool* pool, DestinationId dest, RestVerb type,
+        return;
+      }
 
-        p->tmp_err = err;
-        p->tmp_res = std::move(res);
-        p->tmp_req = std::move(req);
-
-        bool queued = sch->queue(
-          RequestLane::CLUSTER_INTERNAL,
-          [p]() mutable {
-            p->promise.setValue(Response{std::move(p->dest), p->tmp_err,
-                                         std::move(p->tmp_res), std::move(p->tmp_req)});
-          });
-        if (ADB_UNLIKELY(!queued)) {
-          p->promise.setValue(Response{std::move(p->dest), fuerte::Error::QueueCapacityExceeded, nullptr, std::move(p->tmp_req)});
-        }
+      p->tmp_err = err;
+      p->tmp_res = std::move(res);
+      p->tmp_req = std::move(req);
+
+      bool queued = sch->queue(RequestLane::CLUSTER_INTERNAL, [p]() mutable {
+        p->promise.setValue(Response{std::move(p->dest), p->tmp_err,
+                                     std::move(p->tmp_req), std::move(p->tmp_res)});
       });
+      if (ADB_UNLIKELY(!queued)) {
+        p->promise.setValue(Response{std::move(p->dest), fuerte::Error::QueueCapacityExceeded,
+                                     std::move(p->tmp_req), nullptr});
+      }
+    });
     return f;
 
   } catch (std::exception const& e) {
-    LOG_TOPIC("236d7", DEBUG, Logger::COMMUNICATION) << "failed to send request: " << e.what();
+    LOG_TOPIC("236d7", DEBUG, Logger::COMMUNICATION)
+        << "failed to send request: " << e.what();
   } catch (...) {
-    LOG_TOPIC("36d72", DEBUG, Logger::COMMUNICATION) << "failed to send request.";
+    LOG_TOPIC("36d72", DEBUG, Logger::COMMUNICATION)
+        << "failed to send request.";
   }
-  return futures::makeFuture(Response{std::string(), Error::ConnectionCanceled, nullptr, nullptr});
+  return futures::makeFuture(
+      Response{std::string(), Error::ConnectionCanceled, nullptr, nullptr});
 }
 
-/// Handler class with enough information to keep retrying
+/// Stateful handler class with enough information to keep retrying
 /// a request until an overall timeout is hit (or the request succeeds)
 class RequestsState final : public std::enable_shared_from_this<RequestsState> {
  public:
   RequestsState(ConnectionPool* pool, DestinationId&& destination, RestVerb type,
                 std::string&& path, velocypack::Buffer<uint8_t>&& payload,
                 Headers&& headers, RequestOptions const& options)
-      : _payload(std::move(payload)),
-        _destination(std::move(destination)),
-        _path(std::move(path)),
-        _headers(std::move(headers)),
+      : _destination(std::move(destination)),
         _options(options),
         _pool(pool),
-        _type(type),
         _startTime(std::chrono::steady_clock::now()),
         _endTime(_startTime + std::chrono::duration_cast<std::chrono::steady_clock::duration>(
-                                  options.timeout)) {}
+                                  options.timeout)) {
+    _tmp_req = prepareRequest(type, path, payload, _options, headers);
```

Comment:
I think we can use std::move here, because `prepareRequest` expects these parameters by value
```suggestion
    _tmp_req = prepareRequest(type, std::move(path), std::move(payload), _options, std::move(headers));
```